### PR TITLE
fix(graphql): GraphQL queries can exceed the rate limit even with x-ratelimit-remaining !== 0.

### LINKED ISSUE #632: GraphQL rate limit errors don't always get retried.
I ran a GH Action that exceeds its primary rate limit, and the first time (https://github.com/jyasskin/spec-maintenance/actions/runs/6581598228/job/17881791340#check-step-7) plugin-throttling called my `onRateLimit` function and retried accordingly, and the second time (https://github.com/jyasskin/spec-maintenance/actions/runs/6581598228/job/17920483150) (on the same commit), it didn't.

It looks like the ones that skip `onRateLimit` have a 200 response code, while earlier runs that weren't using plugin-throttling showed a 403 response code for secondary rate limits. (They didn't get far enough to exceed the primary rate limit, so I don't have their response code for that.) Could Bottleneck only be firing the `failed` event on HTTP-level failures, and so https://github.com/octokit/plugin-throttling.js/blob/c4147c9d6e54c49aeb7239812da791f2db69858c/src/wrap-request.ts#L44-L54 should be calling `onRateLimit` itself?

### PR DESCRIPTION
**Pull request by jyasskin**
This tries to fix #632, but I haven't tested it.

### CODE REVIEW

**Review by gr2m — COMMENTED** [review]
Thank you so much for investigating the problem and for sending the pull request. We will need a test for it, and a replicable test case to confirm the change works as expected

**Comment by jyasskin** [comment]
I've adjusted the tests so they cover this case. Is https://github.com/jyasskin/testing/actions/runs/6630237322/job/18011241976 what you meant by a replicable test case?

File: test/retry.test.ts [hunk 1]
```diff
@@ -250,7 +250,7 @@ describe("Retry", function () {
             {
               status: 200,
               headers: {
-                "x-ratelimit-remaining": "0",
+                "x-ratelimit-remaining": "1",
```
Comment (gr2m) [inline test/retry.test.ts:253]:
```suggestion
                // should retry based on response body, not header
                // https://github.com/octokit/plugin-throttling.js/issues/632
                "x-ratelimit-remaining": "1",
```

**Review by gr2m — APPROVED** [review]
Thank you, that works!

**Pull request merged**